syft/frameworks/torch/he/fv/context.py

```diff
@@ -13,27 +14,88 @@
 
 
 class Context:
-    """A class used as for holding and easily supplying of all the general
-    parameters required throughout the implementation.
+    def __init__(self, params):
+        """A class used as for holding and easily supplying of all the general
+        parameters required throughout the implementation.
+
+        Args:
+            An EncryptionParams object with polynomial modulus, coefficient modulus and
+            plain modulus set.
+        """
+        # Validation of params provided for the encryption scheme.
+        self.key_param_id = params.param_id
+        self.context_data_map = {}
+        self.context_data_map[self.key_param_id] = ContextData(params)
+
+        if len(params.coeff_modulus) > 1:
+            self.first_param_id = self.create_next_context_data(self.key_param_id)
+        else:
+            self.first_param_id = self.key_param_id
+        self.last_param_id = self.first_param_id
+
+        prev_parms_id = self.first_param_id
+        while len(self.context_data_map[prev_parms_id].param.coeff_modulus) > 1:
+            prev_parms_id = self.create_next_context_data(prev_parms_id)
+            self.last_param_id = prev_parms_id
+
+    def create_next_context_data(self, prev_parms_id):
+        """Generate next context data in context chain by dropping the last coefficient modulus.
+
+        Args:
+            prev_params_id: Id of previous context data in context chain.
+        Returns:
+            Id of newly generated context data in context chain.
+        """
+        next_parms = copy.deepcopy(self.context_data_map[prev_parms_id].param)
+        next_coeff_modulus = next_parms.coeff_modulus[:-1]
+        next_parms.set_coeff_modulus(next_coeff_modulus)
+        next_param_id = next_parms.param_id
+
+        assert next_param_id != prev_parms_id
+
+        next_context_data = ContextData(next_parms)
+        self.context_data_map[next_param_id] = next_context_data
+        self.context_data_map[prev_parms_id].next_context_id = next_param_id
+        self.context_data_map[next_param_id].prev_context_id = prev_parms_id
+        return next_param_id
+
+
+class ContextData:
+    """A class used to hold parameters and perform parameters validation.
+
+    Args:
+        param: An EncryptionParams object with polynomial modulus, coefficient modulus and
+        plain modulus set.
+        prev_context_id: Id of the previous ContextData object in context chain.
+        next_context_id: Id of the next ContextData object in context chain.
```

Comment:
Shouldn't the Args be in the init method?

Reply:
I put the whole doc component inside __init__